### ISSUE TITLE
Journey to Un'Goro Update

### ISSUE THREAD

**Issue by webadict** [demilich1/metastone#333]
Oh ho ho! Guess who's here with an update on Journey to Un'Goro?!

So, there are a BUNCH of changes that need to be made for just the few cards I've seen, and I plan to go over a vast majority of them right now. Bare in mind that this will be a list of all changes that are planned for v1.4.0, and if there's an issue or a change you'd like to see, you're free to leave a comment here.

Biggest Changes:
- PERMANENTS (80% Done!) Cards like Nether Portal and Sherazin, Corpse Flower now leave permanent aspects on the field. This is a bit difficult to program, honestly, but I plan to extend these together with Minions into an abstract Summon class. (Actor -> Summon -> (Minion, Permanent)) These cards will be literally indestructable and untargetable, which means they'll be super fun to customize!
- QUESTS (15% Done!) Each class will have a legendary quest for them to complete. I'm not entirely sure of specifics on this one just yet, but they will be programmed similarly to Secrets, since they do pretty much the same thing. Now, this one is also very tough to set up, since quests have values that countdown from triggers.
- ADAPT (0% Done!) Adapt is a multi-keyword effect, where you gain an aspect that is discovered from 10 different aspects. Good thing we made the Kazakus spell customizable! We can modify that to make it pretty similar.
Big Changes:
- Journey to Un'Goro cards (7 Completed/16 Revealed/135 Total). I'll be slowly making these cards as they're revealed.

That's all I have for now, but this will be modified as more information appears. I'll push some updates as soon as I get Permanents and Quests working, but it is very, very, VERY likely that they will need to be tested to ensure that nothing breaks.

**Comment by mayuso** [comment]
MIght be interesting to show the progress as a checklist:

- [x] This is done
- [ ] This isn't done.

You are doing an incredible job and never stop amazing us, we would love to keep up with your progress and help as much as possible, but it's harder than you think xD I seriously believe a checklist will help both us (those that want to help) and other users that want to know how the changes are going.

This is just my opinion, feel free to ignore it if it makes no sense.

**Comment by webadict** [comment]
Yeah, that might be a good idea. I'll throw it up soon.

**Comment by webadict** [comment]
It took forever, but Quests... Are FINALLY working, though I think they might need to be implemented a bit better.

**Comment by DominusMaximus** [comment]
I received this error when I tried to run your latest build:

> Metastone\game\src\main\java\net\demilich\metastone\game\spells\trigger\secrets\Secret.java:27: error: incompatible types: net.demilich.metastone.game.spells.trigger.secrets.Secret cannot be converted to net.demilich.metastone.game.spells.trigger.types.Secret                event.getGameContext().getLogic().secretTriggered(owner, this);

**Comment by webadict** [comment]
Yeah, I noticed that while uploading. I moved the secrets to their own folder, but it didn't update. I'll upload a copy with working Quests today.

**Comment by webadict** [comment]
See if the new one works. It looks like it _should_ be fixed.

**Comment by DominusMaximus** [comment]
It's running now. 

To see the new cards I had to update the CardSet.java file. 

Other than that an error pops up for The Marsh Queen:

> 22:25:35 ERROR - java.lang.IllegalArgumentException: No enum constant net.demilich.metastone.game.spells.desc.filter.Operation.EQUALS

I'll be testing out the new cards in the coming days :-)

**Comment by webadict** [comment]
Yeah, that's my most up to date copy, so enjoy it. I'm pushing Permanents to @demilich1, so that he doesn't get bombarded by 20 different changes at once.

@demilich1: I never got around to making those test cases... My bad. It took like 45 minutes to get these changes separated from the rest.

**Comment by DominusMaximus** [comment]
Should I post any issues to a new thread, or to this one?

**Comment by webadict** [comment]
Either this thread or post them in my local branch's issue thread.

**Comment by DominusMaximus** [comment]
I can't reach the issues page of your branch. Does this apply to you? 

http://stackoverflow.com/questions/16406180/is-there-a-way-to-add-issues-to-a-github-forked-repo-without-modifying-the-orig

1.) The Charge attribute isn't working.

2.) Errors for Living Mana, Ozruk, Molten Blade and Shifter Zerus popped up upon running the latest build. I attached a file with them. 

[errors.txt](https://github.com/demilich1/metastone/files/877255/errors.txt)

**Comment by webadict** [comment]
lol, yeah, that was my issue.

**Comment by webadict** [comment]
Alright, not sure why ValueProvider isn't working for you, but I fixed a lot of issues. Charge is fixed, Secrets are fixed, and a few other errors are fixed.

**Comment by webadict** [comment]
Alright, next push will be Quests, later tonight.

**Comment by webadict** [comment]
Quest PR is now in review. I'll start pushing cards in the next PR, along with a few bug fixes.

**Comment by webadict** [comment]
Added Adapt! It's currently in my branch. I'll push something eventually.

Added GROUP cards, so that you can literally group specific effects together. Right now, each "card" has to be defined, but eventually, this will be expanded (Such as Spare Parts.)

**Comment by webadict** [comment]
As far as I can tell, I'll probably have all the cards done either this weekend or next weekend, but I won't have time to work on them until the weekend.

**Comment by webadict** [comment]
Added around 1/3 of the remaining cards. Will add more tomorrow.

**Comment by y0urMom** [comment]
When metastone 4.0 comes out, how can I keep modified/custom cards? And to install it, should I just run the installer again or do something ealse?

**Comment by Watermelon86** [comment]
@y0urMom I highly recommend getting an account on Pastebin.com You can store all your custom cards there and download them all whenever you more to a new computer or if you accidently delete or mess up one.

**Comment by TarCreeper** [comment]
@webadict Congratulations for the work, but when the 1.4.0 will exit? Thanks :)
